lib/Models/CatalogNew.ts

```diff
@@ -17,13 +17,28 @@ export default class Catalog {
   constructor(terria: Terria) {
     this.terria = terria;
     this.group = new CatalogGroup("/", this.terria);
+    this.terria.addModel(this.group);
+
+    // create user added data group
+    this.userAddedDataGroup;
   }
 
   get userAddedDataGroup(): CatalogGroup {
     let group = this.userAddedDataGroupIfItExists;
     if (isDefined(group)) {
       return group;
     }
+
+    // check for the case where user added data group exists but was
+    // removed from this.group.memberModels after catalog members were reset
+    group = <CatalogGroup | undefined>(
+      this.terria.getModelById(BaseModel, USER_ADDED_CATEGORY_ID)
+    );
+    if (isDefined(group)) {
+      this.group.add(CommonStrata.definition, group);
+      return group;
+    }
+
```

Comment:
Can you explain what the sequence of events that happens for the user added data group to be in the terria model by ID, but not linked to from `catalog`? Is it when that group is serialised and de-serialised?

If so I think it'd be better to either serialise the user-added data group separately (put it into a different property of the `initData`) or deal with this merging during de-serialisation.

Reply:
Ah yeah so the problem is that when you use updateModelFromJson on members, e.g. in https://github.com/TerriaJS/terriajs/blob/mobx/lib/Models/Terria.ts#L558 all the existing members of the catalog group (including the user added data group) are removed. Probably something which needs to be fixed sometime, but in the meantime I added this to make things at least work. Not too sure what the best thing to do is, would it be better to just go and replace all the relevant occurrences of updateModelFromJson?